overlord/snapstate/handlers_components.go

```diff
@@ -683,6 +685,46 @@ func (m *SnapManager) doSetupKernelModules(t *state.Task, finalStatus state.Stat
 	return nil
 }
 
+func componentsToKeep(t *state.Task, compsup *ComponentSetup, snapsup *SnapSetup, snapst *SnapState) ([]*snap.ComponentSideInfo, error) {
```

Comment:
this is really not pleasant, also remove-kernel-modules-setup could be used in various places, so we need to extra careful what it all means?